runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9326,20 +9326,14 @@ J9::Z::TreeEvaluator::VMmonentEvaluator(TR::Node * node, TR::CodeGenerator * cg)
 
    if (reserveLocking)
       {
-
-      // TODO - ScratchRegisterManager Should Manage these temporary Registers.
-      if (wasteReg)
-         cg->stopUsingRegister(wasteReg);
       cg->stopUsingRegister(monitorReg);
       // TODO : objectClassReg contains the J9Class for object which is set in lwOffset <= 0 case. Usually that is NULL in the following function call
       return reservationLockEnter(node, lwOffset, objectClassReg, cg, helperLink);
       }
 
-   if (normalLockWithReservationPreserving)
```

Comment:
Did we delete this code  or moved allocation of lockPreservingRegister later ?

Reply:
I used scratch register for lockPreservinReg in line 9439.